utility_network/configure-subnetwork-trace/src/main/java/com/esri/samples/configure_subnetwork_trace/ConfigureSubnetworkTraceController.java

```diff
@@ -0,0 +1,424 @@
+      } else {
+        new Alert(Alert.AlertType.WARNING, "No valid comparison value entered").show();
+        return;
+      }
+
+      // create the utility network attribute comparison expression using the specified parameters
+      // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+      UtilityTraceConditionalExpression expression =
+              new UtilityNetworkAttributeComparison(selectedAttribute, selectedOperator, otherValue);
+
+      // check if an expression is already defined for the traversability barriers
+      if (utilityTraversability.getBarriers() instanceof UtilityTraceConditionalExpression) {
+        UtilityTraceConditionalExpression otherExpression =
+                (UtilityTraceConditionalExpression) utilityTraversability.getBarriers();
+        // use the existing expression to create an `or` expression with the user-defined expression
+        expression = new UtilityTraceOrCondition(otherExpression, expression);
+      }
+
+      // set the new expression to the traversability's barriers
+      utilityTraversability.setBarriers(expression);
+
+      // show the expression in the text area
+      traceConditionsTextArea.setText(expressionToString(expression));
+
+    } catch (Exception e) {
+      new Alert(Alert.AlertType.ERROR, "Error adding comparison value.").show();
+    }
+  }
+
+  /**
+   * Parses a utility trace conditional expression into text and returns it.
+   *
+   * @param expression a UtilityTraceConditionalExpression
+   * @return string representing the expression
+   */
+  private String expressionToString(UtilityTraceConditionalExpression expression) {
+
+    StringBuilder stringBuilder = new StringBuilder();
+
+    // for category comparison expressions, add the category name and comparison operator
+    if (expression instanceof UtilityCategoryComparison) {
+      UtilityCategoryComparison categoryComparison = (UtilityCategoryComparison) expression;
+      stringBuilder.append(
+          String.format("'%1$s' %2$s", categoryComparison.getCategory().getName(),
+              categoryComparison.getComparisonOperator().name()));
+    }
+
+    // for network attribute comparison expressions, add the network attribute name and comparison operator
+    else if (expression instanceof UtilityNetworkAttributeComparison) {
+      UtilityNetworkAttributeComparison attributeComparison = (UtilityNetworkAttributeComparison) expression;
+
+      UtilityNetworkAttribute utilityNetworkAttribute = attributeComparison.getNetworkAttribute();
+
+      stringBuilder.append(
+          String.format("'%1$s' %2$s", utilityNetworkAttribute.getName(),
+              attributeComparison.getComparisonOperator().name()));
+
+      if (utilityNetworkAttribute.getDomain() instanceof CodedValueDomain) {
+        CodedValueDomain codedValueDomain = (CodedValueDomain) utilityNetworkAttribute.getDomain();
```

Comment:
Extra comments about how we handle the Domain types might help. For example, here we could say something like:
> For CodedValueDomains, add the name of a CodedValue that matches the network attribute's comparison value (if any)

and below, at the start of the associated `else` block, something like:
> For other types of domain, add the name of the "other" attribute in the comparison, or the value in the comparison if there's no "other" attribute